### PR TITLE
Basil migration to the new translations approach

### PR DESCRIPTION
**Pull request by Trimax**
This PR contains changes to the Basil study.
1. I went through conf files under studies/basil folder and change their structure according to the translation mechanism implemented by Eugene.
2. I added translation mechanism for matrix-type question (according to the suggestion of Yuri)
3. Fixed an issue with nullable variables array

What needs to be done.
Since now I have the confirmation that I'm moving in the right direction, I'm going to process the rest of Basil study conf files and move the definitions to i18n language files.

**How to run**
org.broadinstitute.ddp.studybuilder.StudyBuilderCli
VM options: -Dconfig.file=../../../config/dev_basil.conf
CL arguments: basil --vars output-config/basil-vars.conf -substitutions studies/basil/subs.conf --process-translations PROCESS_IGNORE_TEMPLATES_WITH_TRANSLATIONS --translations-to-db-json

### CODE REVIEW

File: study-builder/studies/basil/loved-one-activity.conf [hunk 1]
```diff
@@ -45,22 +45,11 @@
...
-              </span>"""
-          }
-        ]
-      }
-    ]
+    "templateText": """<span class="ddp-block-title-bold">
+                                  Thank you for submitting your survey. If you would like to make any changes, please reach out to the study team
+                                  at <a href="mailto:info@basilproject.org" class="Footer-contactLink">info@basilproject.org</a>, or call us
+                                  at <a href="tel:857-500-6264" class="Footer-contactLink">857-500-6264</a>.
+                                  </span>""",
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/loved-one-activity.conf]: Let's externalize strings so we can translate to English, Russian, Spanish, etc.

File: study-builder/studies/basil/loved-one-activity.conf [hunk 2]
```diff
@@ -115,33 +75,11 @@
...
-                "name": "instruction_body",
-                "translations": [
-                  {
-                    "language": "en",
-                    "text": "Please fill out as much as you can. All questions are optional. You can return at any time with the link sent to you by email."
-                  }
-                ]
-              }
-            ]
+            "templateText": "<p class=\"PageContent-text\">Please fill out as much as you can. All questions are optional. You can return at any time with the link sent to you by email.</p>",
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/loved-one-activity.conf]: Same as above. Please externalize string and use template strings. Everywhere else too.

**Comment by bskinner** [comment]
This PR needs more details added- there is no information about it in the description aside from the default template.

File: study-builder/studies/basil/loved-one-activity.conf (hunk 1 above)
Comment (Trimax) [reply study-builder/studies/basil/loved-one-activity.conf]: Okay, will do

File: study-builder/studies/basil/loved-one-activity.conf (hunk 2 above)
Comment (Trimax) [reply study-builder/studies/basil/loved-one-activity.conf]: Ok

**Comment by MocanaAtBroad** [comment]
Could you indicate in the comments for this PR how to run the studybuilder cli, i.e. what is the command line to load this configuration?
And do review https://github.com/broadinstitute/ddp-study-server/tree/develop/study-builder/readme.md to see if there are updates needed to that file.

**Comment by Trimax** [comment]
> 

Done

File: study-builder/studies/basil/i18n/ru.conf [hunk 3]
```diff
@@ -0,0 +1,884 @@
...
+  "prequal": {
+    "name": "Prequal",
+    "title": "Заголовок: Prequal",
+    "join_prompt": "Каким образом Вы хотите принимать участие в исследовании?",
+    "join_option_participant": "Я бы хотел принять участие в качестве испытуемого",
+    "join_option_mailing_list": "Я бы хотел подписаться",
+    "join_hint_req": "Пожалуйста, выберите один из вариантов выше",
+  },
+  "autocomplete": {
+    "name": "Picklist Autocomplete",
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/i18n/ru.conf]: Don't feel obligated to translate to Russian, but could you at least at the [ru] prefix for all the strings? Just want to make sure that when we see the app in action through the GUI we don't have to wonder whether the translation correctly by the software or not; if we see text in Russian or with the [ru] prefix we will know there is not bug.

**Review by MocanaAtBroad — COMMENTED** [review]
Let's make sure we have at least a prefix for all the translations strings and then we can move forward.

**Review by ssettipalli — DISMISSED** [review]

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**